pkgs/by-name/ba/bark-server/package.nix

```diff
@@ -0,0 +1,58 @@
+{
+  lib,
+  buildGoModule,
+  fetchFromGitHub,
+  versionCheckHook,
+}:
+
+buildGoModule rec {
+  pname = "bark-server";
+  version = "2.1.5";
+
+  src = fetchFromGitHub {
+    owner = "Finb";
+    repo = "bark-server";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-lPhRWMR+gConFOxiSONBRrwPoARaGqzaIU3hd3wSO8E=";
+    # populate values that require us to use git. By doing this in postFetch we
+    # can delete .git afterwards and maintain better reproducibility of the src.
+    leaveDotGit = true;
+    postFetch = ''
+      cd "$out"
+      git rev-parse HEAD > $out/COMMIT
+      # '0000-00-00T00:00:00Z'
+      date -u -d "@$(git log -1 --pretty=%ct)" "+%Y-%m-%dT%H:%M:%SZ" > $out/SOURCE_DATE_EPOCH
+      find "$out" -name .git -print0 | xargs -0 rm -rf
+    '';
```

Comment:
I'm not against it but it may be over complicated. Usually people don't really care about those commit hash because we already have the version.

Reply:
This is indeed a bit complicated. If it's just `commitHash`, I may not consider adding these and use `${src.rev}`, but the build information input also includes `build time`. For the upstream of comfot, due to the use of `SOURCE_DATE_EPOCH`, adding `commitHash` is also not a big case.

```bash
date -u -d "@$(git log -1 --pretty=%ct)" "+%Y-%m-%dT%H:%M:%SZ" > $out/SOURCE_DATE_EPOCH
```

Inspiration: https://github.com/NixOS/nixpkgs/blob/5358d76fd3967a3ee6f9cee644500f9e6c73e881/pkgs/tools/admin/bom/default.nix#L19

Reply:
It may be beneficial to add some hooks to it. There seems to be a r13y standard.

Reply:
https://github.com/NixOS/nixpkgs/pull/256270